File: python/src/canPlaceFlowers_test.py
from typing import List

import pytest

from canPlaceFlowers import Solution

solution = Solution()


@pytest.mark.parametrize(
    "flowerbed, n, want",
    (
        ([1, 0, 0, 0, 1], 1, True),
        ([1, 0, 0, 0, 1], 2, False),
        ([0, 1, 1, 0, 0], 1, True),
    ),
)
def test_canPlaceFlowers(flowerbed: List[int], n: int, want: bool):
    got = solution.canPlaceFlowers(flowerbed, n)
    assert got == want

File: c/src/canPlaceFlowers_test.c
#include <assert.h>
#include <stdbool.h>
#include <stdlib.h>

#include "canPlaceFlowers.c"

typedef struct test {
    int* flowerbed;
    size_t flowerbedSize;
    int n;
    bool want;
} test;

int main(int argc, char* argv[])
{
    test tests[] = {
        { (int[]) { 1, 0, 0, 0, 1 }, 5, 1, true },
        { (int[]) { 1, 0, 0, 0, 1 }, 5, 2, false },
        { (int[]) { 0, 1, 1, 0, 0 }, 5, 1, true },
    };
    size_t len_tests = sizeof(tests) / sizeof(test);

    for (size_t i = 0; i < len_tests; i++) {
        test* t = &tests[i];
        bool got = canPlaceFlowers(t->flowerbed, t->flowerbedSize, t->n);
        assert(got == t->want);
    }

    exit(EXIT_SUCCESS);
}
